webapp/src/apis/api-helper-cviz.ts

```diff
@@ -0,0 +1,105 @@
+import toast from 'react-hot-toast'
+import EnvironmentVars from '../EnvironmentVars'
+
+class CvizApiHelper {
+  formatQueryParams(query_params?: Record<string, any>): string {
+    if (!query_params || Object.keys(query_params).length === 0) return ''
+    return `?${new URLSearchParams(query_params).toString()}`
+  }
+
+  async invokeApi({
+    path,
+    basePath,
+    method = 'GET',
+    headers = {},
+    queryParams,
+    body,
+    token,
+    timeout,
+    responseType = 'json',
+    booleanResponse = false,
+    toastOnFailure = true,
+    toastOnSuccess = false,
+    successMessage = 'Successfully completed request!',
+    failureMessage = 'Request failed to complete',
+  }: {
+    path: string
+    basePath?: string
+    method?: string
+    headers?: Record<string, string>
+    queryParams?: Record<string, string>
+    body?: Object
+    token?: string
+    timeout?: number
+    responseType?: string
+    booleanResponse?: boolean
+    toastOnFailure?: boolean
+    toastOnSuccess?: boolean
+    successMessage?: string
+    failureMessage?: string
+  }): Promise<any> {
+    const url = (basePath ?? EnvironmentVars.CVIZ_API_SERVER_URL!) + path + this.formatQueryParams(queryParams)
+
+    const localHeaders: HeadersInit = { ...headers }
+    if (token) localHeaders['Authorization'] = `Bearer ${token}`
+    if (method === 'POST' && body && !('Content-Type' in localHeaders)) {
+      localHeaders['Content-Type'] = 'application/json'
+    }
+
+    let controller: AbortController | undefined = undefined
+    let id: NodeJS.Timeout | undefined = undefined
+    if (timeout) {
+      controller = new AbortController()
+      id = setTimeout(() => controller?.abort(), timeout)
+    }
+
+    const options: RequestInit = {
+      method: method,
+      headers: localHeaders,
+      body: body
+        ? localHeaders['Content-Type'] === 'application/x-www-form-urlencoded'
+          ? (body as string)
+          : JSON.stringify(body)
+        : undefined,
+      mode: 'cors',
+      signal: controller?.signal,
+    }
+
+    console.debug('MAKING REQUEST TO ' + url + ' WITH OPTIONS', options)
```

Comment:
In console log statements, arguments separated by a comma are automatically spaced out from the other arguments, and printed next to them. So this would print: "MAKING REQUEST TO URL WITH OPTIONS {...options}"